src/include/kdbtypes.h

```diff
@@ -101,14 +105,14 @@ typedef unsigned int              kdb_unsigned_long_t;
 
 // typedef wchar_t                   kdb_wchar_t; // wchar_t not supported!
 
-#if SIZEOF_LONG == 8
+#if (LONG_MAX == +9223372036854775807)
 #define ELEKTRA_LONG_LONG_F "%ld"
 #define ELEKTRA_LONG_LONG_S strtol
 typedef long                      kdb_long_long_t;
 #define ELEKTRA_UNSIGNED_LONG_LONG_F "%lu"
 #define ELEKTRA_UNSIGNED_LONG_LONG_S strtoul
 typedef unsigned long             kdb_unsigned_long_long_t;
-#elif defined(HAVE_SIZEOF_LONG_LONG) && (SIZEOF_LONG_LONG == 8)
```

Comment:
SIZEOF_LONG_LONG is not defined on my system (OS X SL).
//Result of TRY_COMPILE
HAVE_SIZEOF_LONG_LONG:INTERNAL=FALSE